raft/src/main/java/org/apache/kafka/snapshot/FileSnapshotReader.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.snapshot;
+
+import org.apache.kafka.raft.OffsetAndEpoch;
+import java.util.Iterator;
+import org.apache.kafka.raft.Batch;
+
+/**
+ * Interface of the snapshot reader
+ */
+public interface FileSnapshotReader<T> extends AutoCloseable, Iterator<Batch<T>> {
```

Comment:
How about naming this interface `SnapshotReader` and renaming the class `SnapshotReader` to `RecordsSnapshotReader`

Reply:
> Excuse the confusion as to what was expected in this change. Hope it is clear now.

I'm really sorry to misunderstand what you mean. I think I know how to do that, okay!